pytorch_lightning/trainer/connectors/optimizer_connector.py

```diff
@@ -43,7 +43,13 @@ def update_learning_rates(self, interval: str, monitor_metrics=None):
             if lr_scheduler['interval'] == interval and current_idx % lr_scheduler['frequency'] == 0:
                 # If instance of ReduceLROnPlateau, we need to pass validation loss
                 if lr_scheduler['reduce_on_plateau']:
-                    monitor_key = lr_scheduler['monitor']
+                    try:
+                        monitor_key = lr_scheduler['monitor']
+                    except KeyError as e:
+                        m = "ReduceLROnPlateau requires returning a dict from configure_optimizers with the keyword " \
+                            "monitor=. For example:" \
+                            "return {'optimizer': optimizer, 'lr_scheduler': scheduler, 'monitor': 'your_loss'}"
+                        raise MisconfigurationException(m)
```

Comment:
you can do `raise MisconfigurationException(m) from e` to preserve the stacktrace better